.github/workflows/production-deploy.yml

```diff
@@ -7,56 +7,39 @@ on:
-          -e DOCKER_REGISTRY='${{ env.DOCKER_REGISTRY }}' \
-          -e ERICA_SSH_KEY='${{ secrets.CI_ERICA_SSH_KEY }}' \
-          -e ERICA_HOST='${{ env.ERICA_HOST }}' \
-          -e REGISTRY_PWD='${{ secrets.REGISTRY_PWD }}' \
-          -e CERT_PIN='${{ secrets.CERT_PIN_PRODUCTION }}' \
-          -e QUEUE_URL='${{ secrets.QUEUE_URL_PRODUCTION }}' \
-          -e ERICA_DATABASE_URL='${{ secrets.ERICA_DATABASE_URL_PRODUCTION }}' \
-          -e KUBECONFIG_BASE64='${{ secrets.CI_KUBECONFIG_B64 }}' \
-          -e NAMESPACE='${{ env.NAMESPACE }}' \
-          -e IMAGE_TAG='${{ env.IMAGE_TAG }}' \
-          ${{ env.DOCKER_REGISTRY }}/steuerlotse_deployment erica
-        shell: bash
-
-      - name: Logout of docker container
-        run: docker logout ${{ env.DOCKER_REGISTRY }}
-
-      - name: Report Deployment
-        uses: satak/webrequest-action@v1.2.3
-        env:
-          IMAGE_TAG: ${{ github.event.inputs.imageTag }}
-        with:
-          url: ${{ secrets.METRICS_DEPLOYMENT_WEBHOOK_URL }}
-          method: POST
-          payload: '{"project": "Erica", "version_identifier": "${{ env.IMAGE_TAG }}", "environment": "production", "link": "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"}'
-          headers: '{"Authorization": "Token ${{ secrets.METRICS_WEBHOOK_TOKEN }}"}'
+  syseleven:
+    uses: ./.github/workflows/reusable-deploy-syseleven.yml
+    with:
+      imageTag: ${{ github.event.inputs.imageTag }}
+      environment: production
+    secrets:
+      DOCKER_REGISTRY: ${{ secrets.DOCKER_REGISTRY }}
+      REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
+      REGISTRY_PWD: ${{ secrets.REGISTRY_PWD }}
+      ERICA_SSH_KEY: ${{ secrets.CI_ERICA_SSH_KEY }}
+      ERICA_HOST: ${{ secrets.ERICA_HOST_PRODUCTION }}
+      CERT_PIN: ${{ secrets.CERT_PIN_PRODUCTION }}
+      QUEUE_URL: ${{ secrets.QUEUE_URL_PRODUCTION }}
+      ERICA_DATABASE_URL: ${{ secrets.ERICA_DATABASE_URL_PRODUCTION }}
+      KUBECONFIG_B64: ${{ secrets.CI_KUBECONFIG_B64 }}
+      METRICS_DEPLOYMENT_WEBHOOK_URL: ${{ secrets.METRICS_DEPLOYMENT_WEBHOOK_URL }}
+      METRICS_WEBHOOK_TOKEN: ${{ secrets.METRICS_WEBHOOK_TOKEN }}
+
+  otc:
+    uses: ./.github/workflows/reusable-deploy-otc.yml
+    with:
+      imageTag: ${{ github.event.inputs.imageTag }}
+      environment: production
+    secrets:
+      DOCKER_REGISTRY: ${{ secrets.DOCKER_REGISTRY }}
+      REGISTRY_USER: ${{ secrets.REGISTRY_USER }}
+      REGISTRY_PWD: ${{ secrets.REGISTRY_PWD }}
+      ERICA_SSH_KEY: ${{ secrets.CI_ERICA_SSH_KEY }}
+      ERICA_HOST: ${{ secrets.ERICA_HOST_OTC_PRODUCTION }}
+      CERT_PIN: ${{ secrets.CERT_PIN_OTC_PRODUCTION }}
+      QUEUE_URL: ${{ secrets.QUEUE_URL_OTC_PRODUCTION }}
+      ERICA_DATABASE_URL: ${{ secrets.ERICA_DATABASE_URL_OTC_PRODUCTION }}
+      DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+      ARGOCD_SERVER: ${{ secrets.ARGOCD_SERVER }}
+      ARGOCD_PIPELINE_PASSWORD: ${{ secrets.ARGOCD_PIPELINE_PASSWORD }}
```

Comment:
I am fine with that as the Erica should work for both systems all the time. Is that the reason why you decided to deploy to production to both systems instead of being able to toggle between those two??

Reply:
Hm, interesting. The thought never crossed my mind. Eventually, we'll want everything to use the OTC erica and we don't want to have to coordinate deployments, so maybe good to learn the habits required to make that work early by always deploying to both? (I.e. backwards-compatible changes only…)